include/seastar/core/semaphore.hh

```diff
@@ -153,7 +156,9 @@ public:
     using exception_factory = ExceptionFactory;
 private:
     ssize_t _count;
+    size_t _outstanding_units = 0;
```

Comment:
This is making the semaphore a little bigger.

Reply:
It's not that simple, we need to move it when it's freshly constructed for example in `do_with`

Reply:
True, but I'm not sure how critical that is.
Can we live with two 32-bit counters?

Reply:
if this is only read by `assert()`, can we guard it with `#ifndef NDEBUG`?

Reply:
We can but scylla runs with asserts enabled even in release build mode so I'm not sure it's enough.  @avikivity what do you think?